lib/datasource/docker/index.js

```diff
@@ -50,28 +52,27 @@ async function getAuthHeaders(registry, repository) {
       apiCheckResponse.headers['www-authenticate']
     );
 
-    const { host } = URL.parse(registry);
-    const opts = hostRules.find({ hostType: 'docker', host }) || {};
-    opts.json = true;
-    if (opts.username && opts.password) {
-      const auth = Buffer.from(`${opts.username}:${opts.password}`).toString(
-        'base64'
-      );
-      opts.headers = { Authorization: `Basic ${auth}` };
+    const hostOpts = hostRules.find({ hostType, url: apiCheckUrl });
```

Comment:
we should use ` { ...hostRules.find({ hostType, url: apiCheckUrl }), json: true }` to be sure hostOpts are not null